clustering/marshalling/protostream/pom.xml

```diff
@@ -45,12 +45,6 @@
         Also contains wildfly-clustering-marshalling-spi implementations using ProtoStream.
     </description>
 
-    <properties>
-        <jdk.max.version>9</jdk.max.version>
-        <!-- This property otherwise causes the java8-test execution to crash -->
-        <modular.jdk.args></modular.jdk.args>
-    </properties>
-
```

Comment:
WDYT about this proposal @bstansberry @pferraro ? JDK8 is enforced during deployment to Nexus so it is not needed anymore, right ?

Reply:
I think this is still necessary.  When running the surefire execution during the test phase using JDK9+ and the java8.home property is set, the forked VM otherwise crashes.
e.g. when running on JDK11, I get the following when running `mvn test` on this module:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (java8-test) on project wildfly-clustering-marshalling-protostream: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/paul/wildfly/infinispan/wildfly/clustering/marshalling/protostream/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/paul/wildfly/infinispan/wildfly/clustering/marshalling/protostream && /usr/lib/jvm/java-8-openjdk/bin/java --add-exports=java.base/sun.nio.ch=ALL-UNNAMED --add-exports=jdk.unsupported/sun.reflect=ALL-UNNAMED --add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED --add-modules=java.se -ea -Duser.region=US -Duser.language=en -XX:MaxMetaspaceSize=512m -jar /home/paul/wildfly/infinispan/wildfly/clustering/marshalling/protostream/target/surefire/surefirebooter14829129568028292918.jar /home/paul/wildfly/infinispan/wildfly/clustering/marshalling/protostream/target/surefire 2021-05-26T12-56-32_004-jvmRun1 surefire6286863381485135034tmp surefire_11683391545194545382tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: The forked VM terminated without properly saying goodbye. VM crash or System.exit called?
[ERROR] Command was /bin/sh -c cd /home/paul/wildfly/infinispan/wildfly/clustering/marshalling/protostream && /usr/lib/jvm/java-8-openjdk/bin/java --add-exports=java.base/sun.nio.ch=ALL-UNNAMED --add-exports=jdk.unsupported/sun.reflect=ALL-UNNAMED --add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED --add-modules=java.se -ea -Duser.region=US -Duser.language=en -XX:MaxMetaspaceSize=512m -jar /home/paul/wildfly/infinispan/wildfly/clustering/marshalling/protostream/target/surefire/surefirebooter14829129568028292918.jar /home/paul/wildfly/infinispan/wildfly/clustering/marshalling/protostream/target/surefire 2021-05-26T12-56-32_004-jvmRun1 surefire6286863381485135034tmp surefire_11683391545194545382tmp
[ERROR] Error occurred in starting fork, check output in log
[ERROR] Process Exit Code: 1
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:669)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:282)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:245)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1183)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1011)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:857)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:957)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:289)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:193)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

```

Reply:
Why do we need to test clustering subsystem on both Modular JDK & JDK8 at the same time @pferraro during test phase?
If we would relax that requirement we might be running WildFly on both JDK types, just not at the same time.
One would need to switch the JDK locally before rebuild. This said I'm proposing to remove build-test-java11 and  build-test-java8 marker files.

Reply:
This specific module generates a multi-release jar when compiled using JDK9+.  For full test coverage, surefire tests can be run against multiple JDKs via https://github.com/jboss/jboss-parent-pom#mr-jar-testing

Reply:
Then it is an intentional requirement. We need to find a solution how it could be executable also on JDK16+